daemon/containerd/image_push_test.go

```diff
@@ -0,0 +1,280 @@
+			requestPlatform: &linuxAmd64,
+		},
+		{
+			name:            "linux/amd64 requested, but not present",
+			check:           candidateNotFound,
+			indexPlatforms:  []platforms.Platform{linuxAmd64, darwinArm64, windowsAmd64},
+			deletePlatforms: []platforms.Platform{linuxAmd64},
+			requestPlatform: &linuxAmd64,
+		},
+
+		// Variant tests
+		{
+			name:  "linux/arm/v5 requested, but not in index",
+			check: candidateNotFound,
+
+			indexPlatforms:  []platforms.Platform{linuxAmd64, linuxArmv7},
+			deletePlatforms: nil,
+			requestPlatform: &linuxArmv5,
+		},
+		{
+			name:  "linux/arm/v5 requested, but not available",
+			check: candidateNotFound,
+
+			indexPlatforms:  []platforms.Platform{linuxArm64, linuxArmv7, linuxArmv5},
+			deletePlatforms: []ocispec.Platform{linuxArmv5},
+			requestPlatform: &linuxArmv5,
+		},
+		{
+			name:  "linux/arm/v7 requested, but not available",
+			check: candidateNotFound,
+
+			indexPlatforms:  []platforms.Platform{linuxArm64, linuxArmv7, linuxArmv5},
+			deletePlatforms: []ocispec.Platform{linuxArmv7},
+			requestPlatform: &linuxArmv7,
+		},
+		{
+			name:  "linux/arm/v7 requested on v7 daemon, but not available",
+			check: candidateNotFound,
+
+			indexPlatforms:  []platforms.Platform{linuxArm64, linuxArmv7, linuxArmv5},
+			deletePlatforms: []ocispec.Platform{linuxArmv7},
+			daemonPlatform:  &linuxArmv7,
+			requestPlatform: &linuxArmv7,
+		},
+		{
+			name:  "linux/arm/v7 requested on v5 daemon, all available",
+			check: singleManifestSelected(linuxArmv7),
+
+			indexPlatforms:  []platforms.Platform{linuxArm64, linuxArmv7, linuxArmv5},
+			deletePlatforms: nil,
+			daemonPlatform:  &linuxArmv5,
+			requestPlatform: &linuxArmv7,
+		},
+		{
+			name:  "linux/arm/v5 requested on v7 daemon, all available",
+			check: singleManifestSelected(linuxArmv5),
+
+			indexPlatforms:  []platforms.Platform{linuxArm64, linuxArmv7, linuxArmv5},
+			deletePlatforms: nil,
+			daemonPlatform:  &linuxArmv7,
+			requestPlatform: &linuxArmv5,
+		},
+		{
+			name:  "none requested on v5 daemon, arm64 not available",
+			check: singleManifestSelected(linuxArmv5),
+
+			indexPlatforms:  []platforms.Platform{linuxArm64, linuxArmv7, linuxArmv5},
+			deletePlatforms: []ocispec.Platform{linuxArm64},
+			daemonPlatform:  &linuxArmv5,
+			requestPlatform: nil,
+		},
+		{
+			name:  "none requested on v7 daemon, arm64 not available",
+			check: singleManifestSelected(linuxArmv7),
+
+			indexPlatforms:  []platforms.Platform{linuxArm64, linuxArmv7, linuxArmv5},
+			deletePlatforms: []ocispec.Platform{linuxArm64},
+			daemonPlatform:  &linuxArmv7,
+			requestPlatform: nil,
+		},
+		{
+			name:  "none requested on v7 daemon, v7 not available",
+			check: singleManifestSelected(linuxArmv5), // Should it fail, because v5 can't be pushed?
+
+			indexPlatforms:  []platforms.Platform{linuxArm64, linuxArmv7, linuxArmv5},
+			deletePlatforms: []ocispec.Platform{linuxArmv7},
+			daemonPlatform:  &linuxArmv7,
+			requestPlatform: nil,
+		},
```

Comment:
Thoughts?

Reply:
`deletePlatforms` are the platforms that are missing locally. So the case is:

Image index supports: linux/arm64, linux/arm/v5 and linux/arm/v7.

Daemon is running on: linux/arm/v7

User has pulled: linux/arm64, linux/arm/v5  (image index minus the `deletePlatforms`).

Should the arm/v5 be pushed (as the daemon can run this image), even though it's not the native variant which the original image supports?